src/praisonai/praisonai/bots/_mirror.py

```
@@ -0,0 +1,130 @@
+                "content": message_text,
+                "timestamp": datetime.now(timezone.utc).isoformat(),
+                "mirror": True,
+                "mirror_source": source_label,
+            }
+            if metadata:
+                # Filter out reserved fields to preserve entry invariants
+                _reserved = {"role", "content", "timestamp", "mirror", "mirror_source"}
+                entry.update({k: v for k, v in metadata.items() if k not in _reserved})
+            
+            return session_mgr._add_mirror_entry_sync(user_id, entry)
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)